src/inmanta/deploy/tasks.py

```diff
@@ -120,41 +120,47 @@ async def execute(self, task_manager: "scheduler.TaskManager", agent: str, reaso
             if intent is None:
                 # Stale resource, can simply be dropped.
                 return
-
-            # Dependencies are always set when calling get_resource_intent_for_deploy
-            assert intent.dependencies is not None
-            # Resolve to exector form
-            version = intent.model_version
-            resource_details = intent.details
-            executor_resource_details: executor.ResourceDetails = self.get_executor_resource_details(version, resource_details)
-
-            # Make id's
-            gid = uuid.uuid4()
-            action_id = uuid.uuid4()  # can this be gid ?
+            # From this point on, we HAVE to call report_resource_state to make the scheduler propagate state
 
             # The main difficulty off this code is exception handling
             # We collect state here to report back in the finally block
 
             # Full status of the deploy,
             # may be unset if we fail before signaling start to the server, will be set if we signaled start
             deploy_result: DeployResult | None = None
-            scheduler_deployment_result: state.DeploymentResult
+            scheduler_deployment_result: state.DeploymentResult = state.DeploymentResult.FAILED
```

Comment:
I remember @sanderr not wanting to initialize this value and to explicitly set it if the deploy were to fail. Like in line 154

Reply:
The question is: where do you catch that failure?

We caught all the ones we expect, but if something unexpected happens, do we prefer to crash the scheduler or report failure and continue

Reply:
I think João may be referring to my preference for style a over style b in the snippet below
```python
# style a
x: int
try:
  do_stuff()
  x = 0
except Exception:
  x = 1

# style b
y: int = 1
try:
  do_stuff()
  y = 0
except Exception:
  pass
```

This is mostly a stylistic preference (semi-functional over updating the value), with a small part of confidence that we can't forget to set it in any of the branches.

But it remains mostly stylistic, and thus subjective, so I won't raise a fuss.